src/agent/proxy.c

```diff
@@ -0,0 +1,135 @@
+#include "libhirte/log/log.h"
+
+#include "agent.h"
+#include "proxy.h"
+
+
+static int proxy_job_method_set_ready(sd_bus_message *m, void *userdata, sd_bus_error *ret_error);
+
+static const sd_bus_vtable proxy_job_vtable[] = {
+        SD_BUS_VTABLE_START(0), SD_BUS_METHOD("Ready", "s", "", proxy_job_method_set_ready, 0), SD_BUS_VTABLE_END
+};
+
+static int proxy_job_method_set_ready(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        const char *status = NULL;
+        int r = sd_bus_message_read(m, "s", &status);
+        if (r < 0) {
+                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_INVALID_ARGS, "Invalid arguments");
+        }
+
```

Comment:
I think we should add some verification here that the reply to the initial request is never sent twice. For example:
```
if (proxy->request_message != NULL) {
  ....
  sd_bus_message_unref(proxy->request_message);
  proxy->request_message = NULL;
}
```